### PR TITLE
Allow Student Feedback

### PR DESCRIPTION
**Pull request by Sephster**
This pull request allows tutors to create assessments where student justification for marks can be shared with students, so they have a better understanding of why the were marked the way they were by their peers.

Changes include:

* Mark assessment feedback comments as visible to students
* Warn students their comments will be visible to their peers
* Allow tutors to moderate comments prior to publication
* Remove most "advanced" feature toggles to clean up the UI
* Add new icons for the assessment actions as could not find originals used
* Prevent further edits to comments once the reports have been published
* Students emailed when comments report is ready. They won't need to login and instead will be provided with a link and a long hash to prevent guessing of reports

### CODE REVIEW

File: src/students/assessments/reports/justification_comments.php [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+            <th>#</th>
+            <th>Comment</th>
+        </tr>
+        <?php foreach ($comments as $index => $comment) : ?>
+        <tr>
+            <td><?= $index + 1 ?></td>
+            <td><?= $comment['justification_text'] ?></td>
+        </tr>
+
+        <?php endforeach; ?>
```
[the remark this answers is not in the document]
Comment (Sephster) [reply src/students/assessments/reports/justification_comments.php:88]: good spot. Removed. Thanks!

File: src/tutors/assessments/create/class_wizardstep_5.php [hunk 2]
```diff
@@ -137,6 +137,14 @@ public function form()
         } ?>
     </div>
 
+    <h2>Feedback</h2>
+
+    <?php if ($this->wizard->get_field('view_feedback') === 'view_feedback') : ?>
```
[the remark this answers is not in the document]
Comment (Sephster) [reply src/tutors/assessments/create/class_wizardstep_5.php:142]: ahh sorry Richard - yeah, can see how this is a little confusing. The class and table do store it as a 1 or 0 but the form itself returns the value of `view_feedback`. In this instance, it is comparing the value from the checkbox which is set in class wizardstep_2, line 245. 

Sorry for the confusion there

File: src/students/assessments/reports/justification_comments.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (richeduni) [reply src/students/assessments/reports/justification_comments.php:88]: Thanks

File: src/tutors/assessments/create/class_wizardstep_5.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (richeduni) [reply src/tutors/assessments/create/class_wizardstep_5.php:142]: Cool. Thanks

**Pull request merged**